lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.ts

```diff
@@ -115,39 +116,65 @@ export class StartProcessCloudComponent implements OnChanges, OnInit, OnDestroy
     processDefinitionSelection: EventEmitter<ProcessDefinitionCloud> = new EventEmitter<ProcessDefinitionCloud>();
 
     processDefinitionList: ProcessDefinitionCloud[] = [];
-    processDefinitionCurrent: ProcessDefinitionCloud;
+    processDefinitionCurrent?: ProcessDefinitionCloud;
     errorMessageId: string = '';
-    processForm: UntypedFormGroup;
     processPayloadCloud = new ProcessPayloadCloud();
     filteredProcesses: ProcessDefinitionCloud[] = [];
-    isLoading = false;
-    isFormCloudLoaded = false;
-    formCloud: FormModel;
     staticMappings: TaskVariableCloud[] = [];
-    resolvedValues: TaskVariableCloud[];
+    resolvedValues?: TaskVariableCloud[];
 
     protected onDestroy$ = new Subject<boolean>();
+
+    isProcessStarting = false;
+    isFormCloudLoaded = false;
+    isFormCloudLoading = false;
     processDefinitionLoaded = false;
-    loading$ = new BehaviorSubject<boolean>(!this.processDefinitionLoaded);
 
-    constructor(
-        private startProcessCloudService: StartProcessCloudService,
-        private formBuilder: UntypedFormBuilder,
-        private processNameCloudPipe: ProcessNameCloudPipe
-    ) {}
+    formCloud?: FormModel;
+    processForm = new FormGroup({
+        processInstanceName: new FormControl('', [
+            Validators.required,
+            Validators.maxLength(this.getMaxNameLength()),
+            Validators.pattern('^[^\\s]+(\\s+[^\\s]+)*$')
+        ]),
+        processDefinition: new FormControl('', [Validators.required, this.processDefinitionNameValidator()])
+    });
+
+    private readonly startProcessCloudService = inject(StartProcessCloudService);
+    private readonly processNameCloudPipe = inject(ProcessNameCloudPipe);
+
+    get isProcessFormValid(): boolean {
+        if (this.hasForm && this.isFormCloudLoaded) {
+            return (this.formCloud ? !Object.keys(this.formCloud.values).length : false) || this.formCloud?.isValid || this.isProcessStarting;
+        } else {
+            return this.processForm.valid || this.isProcessStarting;
+        }
```

Comment:
else block is not needed in this case

Reply:
Are you sure about it? I took the original implementation and only refactored it to getter. As mentioned before, I don't want to introduce any breaking change.

Reply:
Actually, we do here if(sth){ return }, else { return } so I'm 100% sure it won't break anything but it's up to you, you can leave it as well

Reply:
(old imp)
![image](https://github.com/user-attachments/assets/a0602831-f6f8-420e-b0a0-28df5044a196)

Reply:
it's code refactor thing

Reply:
Ah, you don't mean removing `return this.processForm.valid || this.isProcessStarting;` but you mean to remove only `else { }`. Sure I could but what for? ;) It's a cosmetic change and as I said already, what I'd love to do here is to create a v2 of this component

Reply:
Sure, maybe it's only my thing that i don't like use else with direct return, resolved